core/src/main/scala/org/apache/spark/status/AppStatusListener.scala

```diff
@@ -223,6 +228,15 @@ private[spark] class AppStatusListener(
     updateNodeBlackList(event.hostId, false)
   }
 
+  def updateBlackListStatusForStage(executorId: String, stageId: Int, stageAttemptId: Int): Unit = {
+    Option(liveStages.get((stageId, stageAttemptId))).foreach { stage =>
+      val now = System.nanoTime()
+      val esummary = stage.executorSummary(executorId)
+      esummary.isBlacklisted = true
+      maybeUpdate(esummary, now)
+    }
+  }
+
```

Comment:
@vanzin you are more familiar with the new history server.  I am wondering why is the updateBlacklistStatus only done with liveUpdate?  Doesn't that mean it won't show up in history server for finished app?

Reply:
LiveEntities periodically write an immutable view of the entity to the store. LiveExecutor and LiveExecutorStageSummary were modified in this PR to maintain blacklisted status and to write it to the state store.

Reply:
`liveUpdate` / `maybeUpdate` are optimizations to avoid unnecessary writes to disk. They can be called for intermediate updates (which would be reflected in a live application), and only force the write at the very last update (so the data is written in the SHS).